covid/question_covid.py

```diff
@@ -142,36 +143,70 @@ def yield_subtext(sentences):
             yield (self.index2paperID[i], best_answer, best_score, best_text, tfidf_score, scibert_score)
 
 def get_data_texts(articles_dir, articles_folders, meta_path):
+
+        def get_abstract_language(abstract):
+            try:
+                language = detect(abstract)
+            except:
+                language = None
+            return language
+
         # Create dict of paper_id and publication year
         meta_data = pd.read_csv(meta_path, low_memory=True)
         paperID2year = {}
+        paperID2lang = {}
+        sha2pmcid = {}
         for _, meta_row in meta_data.iterrows():
-            # The paper ID will either be the pmcid or sha
-            if pd.notnull(meta_row['pmcid']):
-                paperID2year[meta_row['pmcid']] = meta_row['publish_time']
-            # There can be muliple sha IDs in the rows
-            if pd.notnull(meta_row['sha']):
-                paper_ids = meta_row['sha'].split('; ')
-                for paper_id in paper_ids:
-                    paperID2year[paper_id] = meta_row['publish_time']
+            # Only save information for meta data with parsed text
+            if meta_row['has_pmc_xml_parse'] or meta_row['has_pdf_parse']:
```

Comment:
ok i think this isn't working as I thought. I was under the impression that if either of these are True then it means that we have a full text file available, i.e. if both of them are false then we wont find a full text json file and therefore it's not worth finding the language or year.

Reply:
(this assumption was from the metadata.readme file
```
* added PMC XML parses; available under metadata file column "has_pmc_xml_parse"
* renamed "has_full_text" metadata column to "has_pdf_parse"
```